src/Generator.cpp

```diff
@@ -1239,45 +1235,67 @@ GeneratorParamInfo::GeneratorParamInfo(GeneratorBase *generator, const size_t si
         }
     };
 
-    std::vector<void *> vi = ObjectInstanceRegistry::instances_in_range(
-        generator, size, ObjectInstanceRegistry::GeneratorInput);
-    for (auto *v : vi) {
-        auto *input = static_cast<Internal::GeneratorInputBase *>(v);
-        internal_assert(input != nullptr);
-        user_assert(is_valid_name(input->name())) << "Invalid Input name: (" << input->name() << ")\n";
-        user_assert(!names.count(input->name())) << "Duplicate Input name: " << input->name();
-        names.insert(input->name());
-        internal_assert(input->generator == nullptr || input->generator == generator);
-        input->generator = generator;
-        filter_inputs.push_back(input);
-        add_synthetic_params(input);
-    }
-
-    std::vector<void *> vo = ObjectInstanceRegistry::instances_in_range(
-        generator, size, ObjectInstanceRegistry::GeneratorOutput);
-    for (auto *v : vo) {
-        auto *output = static_cast<Internal::GeneratorOutputBase *>(v);
-        internal_assert(output != nullptr);
-        user_assert(is_valid_name(output->name())) << "Invalid Output name: (" << output->name() << ")\n";
-        user_assert(!names.count(output->name())) << "Duplicate Output name: " << output->name();
-        names.insert(output->name());
-        internal_assert(output->generator == nullptr || output->generator == generator);
-        output->generator = generator;
-        filter_outputs.push_back(output);
-        add_synthetic_params(output);
-    }
-
-    std::vector<void *> vg = ObjectInstanceRegistry::instances_in_range(
-        generator, size, ObjectInstanceRegistry::GeneratorParam);
-    for (auto *v : vg) {
-        auto *param = static_cast<GeneratorParamBase *>(v);
-        internal_assert(param != nullptr);
-        user_assert(is_valid_name(param->name())) << "Invalid GeneratorParam name: " << param->name();
-        user_assert(!names.count(param->name())) << "Duplicate GeneratorParam name: " << param->name();
-        names.insert(param->name());
-        internal_assert(param->generator == nullptr || param->generator == generator);
-        param->generator = generator;
-        filter_generator_params.push_back(param);
+    const char *const io_order_warning = "Generators will always produce code that orders all Inputs before all Outputs; "
```

Comment:
The warning should probably include instructions to override that method if you explicitly want to have outputs declared before inputs

Reply:
Not sure if I entirely agree with this -- part of me wants to push back and say that anyone who *really* needs it will figure it out. But I'm probably overthinking it...